### PR TITLE
Swift 2.0

### PR DESCRIPTION
**Pull request by wvteijlingen**

### CODE REVIEW

**Comment by kurko** [comment]
:tada: :tada: :tada: :tada: :tada:

**Comment by kurko** [comment]
I'm getting the following error when trying to compile my app with this branch: https://gist.github.com/kurko/560ccaf4df5207f10193

Do you have any idea what that is? Perhaps related to ResourceProtocol?

**Comment by wvteijlingen** [comment]
Are you using Cocoapods or Carthage here?

**Comment by kurko** [comment]
Cocoapods.

**Comment by wvteijlingen** [comment]
It seems to be some kind of problem with the Result dependency that BrightFutures needs. In any case, I removed the Future extensions to slim down this library.

**Comment by kurko** [comment]
@wvteijlingen :heart: gonna retry tonight.

File: README.md [hunk 1]
```diff
@@ -87,19 +84,33 @@ spine.registerResource(Post.resourceType) { Post() }
 ```
 
 ### 3. Fetching resources
+
+#### Using find methods
 ```swift
-// Using simple find methods
-spine.find(["1", "2"], ofType: Post.self) // Fetch posts with ID 1 and 2
+// Fetch posts with ID 1 and 2
+spine.find(["1", "2"], ofType: Post.self).onSuccess { resources, meta, jsonapi in
```
Comment (kurko) [inline README.md]: I love this.

**Comment by kurko** [comment]
@wvteijlingen :heart: gonna retry tonight.

**EDIT:** YAY! It worked! I was able to get my application to run in Swift 2 with Xcode 7 and this branch.

Only problem I have now is I am getting an error with spine.HTTPClient. I can't set authorization headers.

screen shot 2015-10-23 at 6 41 04 pm

File: Spine/Networking.swift [hunk 2]
```diff
@@ -8,35 +8,43 @@
...
+		return request(method, URL: URL, payload: nil, callback: callback)
+	}
 }
 
 /**
-The built in HTTP client that uses an NSURLSession for networking.
+The HTTPClient implements the NetworkClient protocol to work over an HTTP connection.
 */
-public class URLSessionClient: _HTTPClientProtocol {
+public class HTTPClient: NetworkClient {
```
Comment (kurko) [inline Spine/Networking.swift]: Researching the code, I found out how to define headers by injecting this into `Spine()`, but somehow HTTPClient is not being accessible. Check this out:

nohttpclient

**Comment by wvteijlingen** [comment]
This should be fixed now. The easiest way to set headers would be like this though:

```swift
(self.networkClient as! HTTPClient).setHeader("header", to: "value")
```

**Comment by kurko** [comment]
Works great! This was the last pending thing. I'll continue working on Xcode7/Swift2 now until I find something else :D

**Comment by kurko** [comment]
Found an issue. I have 2 default environments (Debug, Release) and another one I created (Staging), according to image below:

screen shot 2015-10-28 at 11 36 07 pm

When I run a POST/PUT request in my mobile device using the Build Configuration `Staging`, I get this:

screen shot 2015-10-28 at 11 34 12 pm

When I use `Debug`, it works fine.

I do `GET`'s just fine. Seems similar to https://github.com/wvteijlingen/Spine/pull/32. Any ideas?

How about doing:

```
URLForResourceType((resource as! Resource).dynamicType.resourceType).URLByAppendingPathComponent("/\(resource.id!)")
```

File: Spine/Routing.swift [hunk 3]
```diff
@@ -60,12 +59,12 @@ public class Router: RouterProtocol {
 		return baseURL.URLByAppendingPathComponent(type)
 	}
 	
-	public func URLForRelationship<T: ResourceProtocol>(relationship: Relationship, ofResource resource: T) -> NSURL {
+	public func URLForRelationship<T: Resource>(relationship: Relationship, ofResource resource: T) -> NSURL {
 		let resourceURL = resource.URL ?? URLForResourceType(resource.dynamicType.resourceType).URLByAppendingPathComponent("/\(resource.id!)")
```
Comment (kurko) [inline Spine/Routing.swift]: Found an issue. I have 2 default environments (Debug, Release) and another one I created (Staging), according to image below:

screen shot 2015-10-28 at 11 36 07 pm

When I run a POST/PUT request in my mobile device using the Build Configuration `Staging`, I get this:

screen shot 2015-10-28 at 11 34 12 pm

When I use `Debug`, it works fine.

I do `GET`'s just fine. Seems similar to https://github.com/wvteijlingen/Spine/pull/32. Any ideas?

I did the following and it worked:

```
URLForResourceType((resource as! Resource).dynamicType.resourceType).URLByAppendingPathComponent("/\(resource.id!)")
```

Comment (kurko) [inline Spine/Routing.swift]: I can add a patch for that if you prefer.

**Pull request merged**